service/src/main/java/org/ehrbase/service/DirectoryServiceImp.java

```diff
@@ -134,23 +136,24 @@ public Folder create(UUID ehrId, Folder folder, UUID contributionId, UUID auditI
 
         // validation
         ehrService.checkEhrExistsAndIsModifiable(ehrId);
-        if (ehrFolderRepository.hasFolder(ehrId, EHR_DIRECTORY_FOLDER_IDX)) {
+        if (ehrFolderRepository.hasFolderForEhr(ehrId, EHR_DIRECTORY_FOLDER_IDX)) {
             throw new StateConflictException("EHR with id %s already contains a directory.".formatted(ehrId));
         }
 
         FolderUtils.checkSiblingNameConflicts(folder);
-
-        updateUuid(
-                folder,
-                true,
-                Optional.ofNullable(folder.getUid())
-                        .map(UIDBasedId::getRoot)
-                        .map(UID::getValue)
-                        .map(UUID::fromString)
-                        .orElse(UuidGenerator.randomUUID()),
-                1);
-
-        ehrFolderRepository.commit(ehrId, folder, contributionId, auditId, EHR_DIRECTORY_FOLDER_IDX);
+        UUID folderUid = Optional.ofNullable(folder.getUid())
+                .map(UIDBasedId::getRoot)
+                .map(UID::getValue)
+                .map(UUID::fromString)
+                .orElse(UuidGenerator.randomUUID());
+
+        updateUuid(folder, true, folderUid, 1);
+
```

Comment:
We need to check that no deleted folder already has the folderUid (check version 1)

Reply:
Already done by the check on the start of the create method
```
if (ehrFolderRepository.hasFolderForEhr(ehrId, EHR_DIRECTORY_FOLDER_IDX)) {
    throw new StateConflictException("EHR with id %s already contains a directory.".formatted(ehrId));
}
```
@vladislavploaia we should add a test for this case if not done yet - Create & Delete a Folder with a specified UID and then try to create them again.

Reply:
This checks the ehr_id, not the vo_id

Reply:
The general flow for deleting and creating is not clear to me. Currently you can delete a folder and when trying to create a new you get a 409 conflict because it check if there is any directory with index 1 in the history - what is true for a deletion. This is not changed with the introduction of the vo_id indexes and checks.

How is it supposed to react in such a case? Should it allow to create new folders with different vo_id in case there is a deleted one?

Reply:
A deleted folder is just a special version "with no content". To recreate it you need to do an update.

Reply:
Yes this is clear - it also means the current implementation of the create method is fine - it does not allow to create a new Folder - regardless of any folder vo_id check. You can, at any point in time, use the update method to re-create the folder.